sql/src/main/java/io/crate/planner/operators/GroupHashAggregate.java

```diff
@@ -90,22 +90,33 @@ public Plan build(Planner.Context plannerContext,
             plan = Merge.ensureOnHandler(plan, plannerContext);
         }
         List<Symbol> sourceOutputs = source.outputs();
-        boolean clusteredByGroupBy = clusteredByGroupKey();
-        if (clusteredByGroupBy ||
-            ExecutionPhases.executesOnHandler(plannerContext.handlerNode(), plan.resultDescription().nodeIds())) {
-
+        boolean shardContainsAllGroupKeyValues = shardsContainAllGroupKeyValues();
+        if (shardContainsAllGroupKeyValues) {
```

Comment:
Could we just move the method call inside the if? AFAIK it is only used once and both the method name and the variable name speak for themselves.

Reply:
+1 for the new name btw!

Reply:
Good point, changed it.